src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,93 @@
+import {useEffect, useState} from "react";
+import {useParams} from "react-router-dom";
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
-import { PeopleTable } from './PeopleTable';
+import {PeopleTable} from "./PeopleTable";
+import {getPeople} from "../api";
+import {Person} from "../types";
+
+function getParentByName(name: string | null, peopleArray: Person[]) {
+  const parent = peopleArray.find(person => person.name === name);
+
+  return parent || null;
+}
 
 export const PeoplePage = () => {
-  return (
-    <>
-      <h1 className="title">People Page</h1>
+  const [isLoading, setIsLoading] = useState(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [hasServerError, setHasServerError] = useState(false);
+  const { personSlug = '' } = useParams();
+
+  const getPeopleFromServer = async () => {
+    setIsLoading(true);
+
+    try {
+      const peopleFromServer = await getPeople();
+      const peopleWithParents = peopleFromServer.map(person => {
+        const personWithParents = {...person};
+        const mother = getParentByName(person.motherName, peopleFromServer);
+        const father = getParentByName(person.fatherName, peopleFromServer);
+
+        if (mother) {
+          personWithParents.mother = mother;
+        }
+
+        if (father) {
+          personWithParents.father = father;
+        }
+
+        return personWithParents;
```

Comment:
More often we collect data for a new field in separate variables and then when making return we spread everything into a new object. This way it's easier to read and change.
```suggestion
        const mother = getParentByName(person.motherName, peopleFromServer);
        const father = getParentByName(person.fatherName, peopleFromServer);

        return {
            ...person,
            ...mother && { mother },
            ...father && { father },
        };
```